src/OmniSharp.LanguageServerProtocol/Handlers/OmniSharpHoverHandler.cs

```diff
@@ -48,7 +45,7 @@ public async override Task<Hover> Handle(HoverParams request, CancellationToken
             {
                 // TODO: Range?  We don't currently have that!
                 // Range =
-                Contents = new MarkedStringsOrMarkupContent(new Container<MarkedString>(Helpers.EscapeMarkdown(omnisharpResponse.Type), Helpers.EscapeMarkdown(omnisharpResponse.Documentation)))
+                Contents = new MarkedStringsOrMarkupContent(new MarkupContent() { Value = omnisharpResponse.Markdown })
```

Comment:
This does imply a bug in the implementation of the QuickInfo provider, as it's not escaping any markdown in the raw text that it appends. The escape helper isn't available at that project level though: is there any other helper available there for escaping markdown?

Reply:
`Regex.Replace(markdown, @"([\\`\*_\{\}\[\]\(\)#+\-\.!])", @"\$1")`? 😄

Reply:
https://microsoft.github.io/language-server-protocol/specifications/specification-3-16/#textDocument_hover

The request description says: `An optional range is a range inside a text document that is used to visualize a hover, e.g. by changing the background color.`

So likely this should be the range of the symbol that's under the cursor.

Reply:
Whats the behavior with this change, it might not need to be escaped (but I'm not sure).

I can probably test this tonight.

Reply:
If there are underscores or stars in the text, they'll appear as markdown. It'll happen here or if you use the standard io version of the provider. I'll make a quick update tonight and add a test.

Reply:
Feel free to move / copy the helper if needed to the QuickInfo side!